typescript/sdk/src/ism/RadixIsmReader.ts

```diff
@@ -0,0 +1,107 @@
+import { RadixSDK, RadixSigningSDK } from '@hyperlane-xyz/radix-sdk';
+import { Address, WithAddress, assert, rootLogger } from '@hyperlane-xyz/utils';
+
+import { ChainMetadataManager } from '../metadata/ChainMetadataManager.js';
+
+import {
+  DerivedIsmConfig,
+  DomainRoutingIsmConfig,
+  IsmType,
+  MultisigIsmConfig,
+  RadixIsmTypes,
+} from './types.js';
+
+export class RadixIsmReader {
+  protected readonly logger = rootLogger.child({
+    module: 'RadixIsmReader',
+  });
+
+  constructor(
+    protected readonly metadataManager: ChainMetadataManager,
+    protected readonly sdk: RadixSDK | RadixSigningSDK,
```

Comment:
nit: could we maybe define an interface that describes which methods are needed for the `RadixIsmReader` class to work instead of specifying the concrete types?

Reply:
There is an IsmReader interface but it requires implemented EVM specific methods which don't exist in other VMs like Cosmos or Radix: https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/chore/radix-ism-reader/typescript/sdk/src/ism/EvmIsmReader.ts#L54

There is one method which is shared by every VM, should I remove all the other methods from the IsmReader interface then to make it accessible for the other Readers?